File: server/src/application/home/get-home-summary.ts
import type { HomeSummaryRepository } from '../../domain/platform/home-summary-repository.js';
import type { HomeSummary } from '../../domain/platform/home-summary.js';

export class GetHomeSummary {
  constructor(private readonly repository: HomeSummaryRepository) {}

  execute(): Promise<HomeSummary> {
    return this.repository.find();
  }
}

File: server/app/application/home/get_home_summary.py
from app.domain.platform.models import HomeSummary
from app.domain.platform.repositories import HomeSummaryRepository


class GetHomeSummary:
    def __init__(self, repository: HomeSummaryRepository) -> None:
        self._repository = repository

    def execute(self) -> HomeSummary:
        return self._repository.find()
